milner_desvars/wk3-notes.md

```diff
@@ -88,31 +51,46 @@ In the project turtlebot, also the launch files are organized in the respective
 
 ### Valgrind
 
-The Valgrind tool suite provides several debugging and profiling tools that help you make your programs faster and more correct.
+The Valgrind tool suite provides several debugging and profiling tools. Nothing from the original gets run directly on the host processor, instead, works this way:
+
+A program translated the code into a temporary simpler form, called Intermediate Representation (IR), after that, a tool (Memcheck for example), can implement changes on the code and introduced its commands to do the necessary tests, after the changes, Valgrind recompiles binary coded to run on a host and other simulated CPUs to execute, and the tool that you are using offers the feedback with the results of the tests.
 
 #### Profiling a node with valgrind while launching ROS
 
 We can attach the diagnosis tool to the node in the launch file adding the following line to the launch file:
 
 `launch-prefix="valgrind"`
 
+Memcheck is the default tool you get when using Valgrind, you can add `--leak-check=yes` to obtain more detailed information about the tests.
+
+To use another tool you the syntax is to add `--tool=<toolname>`.
+
+Some of these tools are:
+
+* memcheck (used by default if you don't put any `--tool`): a memory error detector.
+* cachegrind: a cache and branch-prediction profiler.
+* callgrind: a call=graph generating cache profiler.
+* helgrind: a thread error detector.
+* drd: another thread error detector.
+
+All of these tools provide a different type of output so you have to be knowledgable in how to interpret the results.
+
 Ex.
 
 ```
-<node name="ca_driver" pkg="ca_driver" type="ca_driver" output="screen">
-    <rosparam command="load" file="$(arg config)" />
-    <param name="robot_model" value="CREATE_2" />
-    launch-prefix="valgrind"
-  </node>
+  <node pkg="ca_move_base" type="send_goal" name="navigation_goals" 
+        args="$(arg x) $(arg y) $(arg Y)" ns="$(arg ns)" launch-prefix="valgrind/>
+</launch>
   ```
 
 ### Example of profiling with Valgrind
 
-First of all we need to install Valgrind, we are doing that with the following command:
+First of all, we need to install the tools that would help us to analyze this node, these are, valgrind to do the profiling, and kchegrind and graphviz to represent them in a graphical interphase:
```

Comment:
Typo: Interface

Please, take some time to check the grammar. We already reviewed this word before.